inference.py

```diff
@@ -305,80 +276,57 @@ def main(params: dict):
-                                  f"Will use the first {num_bands} bands of the input image.")
-                    inf_sample['sat_img'] = inf_sample['sat_img'][:, :, 0:num_bands]
-                    print(f"Input image's new shape: {inf_sample['sat_img'].shape}")
-
-                elif input_band_count < num_bands:
-                    warnings.warn(f"Skipping image: The number of bands requested in the yaml file ({num_bands})"
-                                  f"can not be larger than the number of band in the input image ({input_band_count}).")
-                    continue
-
-                # START INFERENCES ON SUB-IMAGES
-                sem_seg_results_per_class = sem_seg_inference(model,
-                                                              inf_sample['sat_img'],
-                                                              nbr_pix_overlap,
-                                                              chunk_size,
-                                                              num_classes_corrected,
-                                                              device,
-                                                              meta_map,
-                                                              inf_sample['metadata'],
-                                                              output_path=working_folder,
-                                                              index=_tqdm.n,
-                                                              debug=debug)
-
-                # CREATE GEOTIF FROM METADATA OF ORIGINAL IMAGE
-                tqdm.write(f'Saving inference...\n')
-                if get_key_def('heatmaps', params['inference'], False):
-                    tqdm.write(f'Heatmaps will be saved.\n')
-                vis(params, inf_sample['sat_img'], sem_seg_results_per_class, working_folder, inference_input_path=local_img, debug=debug)
-
-                tqdm.write(f"\n\nSemantic segmentation of image {img_name} completed\n\n")
-                if bucket:
-                    bucket.upload_file(inference_image, os.path.join(working_folder, f"{img_name.split('.')[0]}_inference.tif"))
-    else:
-        raise ValueError(
-            f"The task should be either classification or segmentation. The provided value is {params['global']['task']}")
-
+                    print(inference_image)                    
+                assert local_img.is_file(), f"Could not locate raster file at {local_img}"
+                with rasterio.open(local_img, 'r') as raster:
+                    inf_meta= raster.meta
+                    if prediction_with_smoothing:
+                        print('Smoothening Predictions with 2D interpolation')
+                        pred = segmentation_with_smoothing(raster, local_gpkg, model, chunk_size, overlap, num_bands, device)
+                    else:
+                        pred = segmentation(raster, local_gpkg, model, chunk_size, num_bands, device)
```

Comment:
without any overlap, I'm afraid inference could be disappointing...

Reply:
Yeah, two versions of prediction functions are offered now. One with overlaps and smoothening and another base version without smoothening.